web/src/components/SubscribeDialog.js

```diff
@@ -77,6 +89,17 @@ const SubscribePage = (props) => {
         console.log(`[SubscribeDialog] Successful login to ${topicUrl(baseUrl, topic)} for user ${username}`);
         props.onSuccess();
     };
+    const generateTopicName = () => {
+        const entropy = randomAlphanumericString();
+        let newTopic = props.topic;
+        if (newTopic) {
+            newTopic += "-" + entropy;
+        } else {
+            const sliceIndex = entropy.length / 2;
+            newTopic = entropy.slice(0, sliceIndex) + "-" + entropy.slice(sliceIndex);
```

Comment:
That seems quite random. Why the dash?

Reply:
Okay I see it now. Didn't see the `contains(-)` before. 

That's ... unexpected and quite the weird UX. IMHO the "append with a dash" logic is also unexpected. A "Generate" button usually just overwrites the existing thing. And then you don't have to ever disable the button. Just make the button generate a random string. Easy as pie.